### ISSUE TITLE
[Live website is not working correctly]

### ISSUE THREAD

**Issue by jpradocueva** [standards-hub/docs#164]
The latest changes, the addition of new floating micro-cards, introduce some stability issues in the live website.
[www](https://temp.openmobilealliance.org/)

The right menus don't move, and the landing page displays some code.

**Comment by FroudeDescartes** [comment]
@jpradocueva  I can see the issue as well. I will inspect it and tag you once I find the solution

**Comment by FroudeDescartes** [comment]
@jpradocueva I can see in the console two errors that would imply that it did not populate the site with content and components as expected (although when run locally, everything works). I would say this is the same thing as in today`s meeting when we had a non-working ToC.

Please, try and generate -> deploy again and see if the results are still the same. If they are, I will investigate further.

**Comment by jpradocueva** [comment]
@FroudeDescartes I generated and deployed the website three times, with mixed results. I will do it once more after I reset my system.

**Comment by FroudeDescartes** [comment]
Resolved with [PR#259](https://github.com/OpenMobileAlliance/oma-knowledge-base/pull/259)

**Comment by FroudeDescartes** [comment]
ToC is still not visible, maybe I will take a different approach in implementing it

**Comment by jpradocueva** [comment]
@FroudeDescartes I run ChatGPT based on the errors and this is what it has provided:

### Issue: Static Page Generation Warnings and Errors During Nuxt Build Process

#### Description:
During the generation of static pages using Nuxt, the following warnings and errors were encountered. This issue seems to be related to multiple 404 errors and excessive chunk sizes. Below is a summary of the key issues:

#### Warnings:
1. **Chunk Size Warning:**
   - [ ]  Some chunks are larger than 500 kB after minification.
   - [ ] Suggested actions:
     - [ ] Use `dynamic import()` to code-split the application.
     - [ ] Modify `build.rollupOptions.output.manualChunks` to improve chunking as outlined in [Rollup's documentation](https://rollupjs.org/configuration-options/#output-manualchunks).
     - Adjust chunk size limit via `build.chunkSizeWarningLimit` in the Nuxt configuration.

2. **Module "stream" Externalization Warning:**
   - [ ] Vite has externalized the `stream` module for browser compatibility.
   - [ ] This warning is triggered by the `sax` library in `node_modules/sax/lib/sax.js`.
   - [ ] Suggested fix: Review the [Vite documentation](https://vitejs.dev/guide/troubleshooting.html#module-externalized-for-browser-compatibility).

#### Errors:
1. **Document Not Found (404) Errors:**
   - Multiple 404 errors were encountered for missing documents during the static page generation process. The following paths are affected:
     - [ ] `/omaspecworks/get-involved/faq`
     - [ ] `/lwm2m/contact-us`
     - [ ] `/lwm2m/community/www.ioterop.com/iowa`
     - [ ]  `/www.ericsson.com/en`
     - [ ]  `/www.ericsson.com/en/portfolio`
     - [ ]  `/www.ericsson.com/en/portfolio/digital-services`
     - [ ]  `/www.ericsson.com/en/portfolio/digital-services/cloud-core`
     - [ ] `/www.ericsson.com/en/portfolio/digital-services/cloud-core/cloud-unified-data-management-and-policy`
   
   - These errors suggest that the linked documents or resources are missing or have incorrect paths.

2. **Linked Documents Issue:**
   - Each 404 error is linked from a specific route, which indicates a missing document or an incorrect reference.
   - Possible causes:
     - [ ]  Missing JSON files in the `api/_content` directory.
     - [ ] Incorrect linking paths in the markdown or component files.
     - [ ] Unresolved paths for dynamic content.

#### Suggested Fixes:
1. **Address Chunk Size Warning:**
   - [ ] Split large components or pages into smaller modules using dynamic imports.
   - [ ] Consider lazy loading non-critical components.

2. **Resolve Missing Documents:**
   - [ ] Verify that the paths in the source code are correctly defined.
   - [ ] Check if the content exists in the `/api/_content` directory.
   - [ ] Create or update the missing documents where applicable.

3. **Correct External Module Reference:**
   - [ ] Investigate the usage of the `sax` library and determine if an alternative approach is feasible for client-side usage.

4. **Skip or Correct Broken Links:**
   - [ ] Update the routes or remove references to missing documents if they are obsolete.

#### Logs:
- [ ] For detailed error and warning logs, please refer to the attached build output from the `nuxt generate` command.

#### Additional Context:
- **Nuxt Version:** 3.12.4
- **Nitro Version:** 2.9.7
- **Vite Version:** 5.4.6

**Comment by jpradocueva** [comment]
@FroudeDescartes, this is the link to another clue as to why the news pages are not displaying on the `temp.` website. See below:


markdown
Copy code
### Issue: Error When Opening News Pages

**Error Message:**
```json
{
  "message": "You should use slots with <ContentRenderer>",
  "value": null,
  "excerpt": false,
  "tag": "div"
}
```
**Description:** The error is triggered when attempting to open a news link stored in the news/articles folder. The issue is likely related to how the ContentRenderer component is being used in the Nuxt application. The ContentRenderer expects content to be passed via slots, and if these slots are not correctly configured or the content is missing, the component will throw this error.

## Why the Error Occurs:
### 1. Missing or Incorrect Slot Usage:

* The ContentRenderer component requires proper usage of slots to insert dynamic content. If no content is provided to these slots, or if they are incorrectly defined, this error will occur.

### 2. Content Formatting Issue:

* The content stored in news/articles may not be properly formatted or passed to the ContentRenderer component. This typically happens if the content object is null or malformed, resulting in the component being unable to render the content.

### 3. Incorrect Component Usage:

* If the <ContentRenderer> component is used without slots or necessary properties, it will fail to render the content and throw this error.

## Suggested Fixes:
### 1. Use Slots Correctly:

Ensure that the ContentRenderer component is used with appropriate slots, as shown in the example below:
```vue
Copy code
<ContentRenderer>
  <template #default="{ content }">
    <div v-html="content"></div>
  </template>
</ContentRenderer>
```
* If using markdown or JSON content, make sure that the content property is properly passed and utilized within the component.

### 2. Check Content Existence:

* Verify that the content being loaded into news/articles is correctly passed to the ContentRenderer. If the content object is null, this could trigger the error.

### 3. Update Content Module Configuration:

* If using Nuxt Content or a custom content module, make sure the configuration is correctly set up to pass content to ContentRenderer as slots. You may need to update the logic to include a check for missing content or implement a fallback in case the content is not yet loaded.

### 4. Debug Content Loading Logic:
* Use console.log or a Vue.js debugging tool to inspect the content being passed to ContentRenderer. Ensure the content is properly formatted and has the necessary properties (e.g., value, excerpt, tag).

## Example Fix in Template:
If you have a template rendering a news article, update it as follows:

```vue
Copy code
<template>
  <ContentRenderer>
    <template #default="{ content }">
      <div v-if="content" v-html="content.value"></div>
      <div v-else>
        <p>Error: Content could not be rendered. Please check the source.</p>
      </div>
    </template>
  </ContentRenderer>
</template>
```
In this example, we are using a default slot to ensure that the ContentRenderer has access to the content property and is able to render it correctly.

## Additional Considerations:
* Review the news/articles folder structure and ensure that all files have consistent frontmatter and structure.
* Consider using a fallback mechanism if the content is not loaded yet (e.g., display a loading spinner or a placeholder message).
By ensuring that the ContentRenderer component has the necessary slot content and the news articles are correctly formatted, this issue should be resolved.

**Comment by FroudeDescartes** [comment]
- [x] This [PR](https://github.com/OpenMobileAlliance/oma-knowledge-base/pull/267) resolves the ToC visibility issue

**Comment by jpradocueva** [comment]
- [x] Pending rendering articles. Currently, the news page opens in production, but not the articles.

**Comment by FroudeDescartes** [comment]
Resolved in [PR#274](https://github.com/OpenMobileAlliance/oma-knowledge-base/pull/274)
